dashboard/app/models/concerns/partial_registration.rb

```diff
@@ -8,16 +8,64 @@ module PartialRegistration
   extend ActiveSupport::Concern
 
   USER_ATTRIBUTES_SESSION_KEY = 'devise.user_attributes'
-  OAUTH_PARAMS_TO_STRIP = %w(oauth_token oauth_refresh_token)
+
+  # A common error during registration, especially when OAuth is involved, is a
+  # Cookie Overflow.  This module stores attributes for a partially-created user
+  # in the _learn_session cookie, which cannot exceed 4KB.  In fact, it we seem
+  # to encounter issues when the cookie is more than 2KB, since Devise and other
+  # parts of our application may also be using the session.
+  # To work around this, we keep known large values out of the session and
+  # instead store them in the shared application cache.
```

Comment:
> Should we put something in place to ensure that no one "thing" getting written to the rails session exceeds a certain size (conservatively, 1KB)? This wouldn't fix everything, but could prevent against future regressions, and/or help isolate failures to problematic features without data loss affecting other parts of the app.

I like the idea of a warning (maybe a Honeybadger notification) for any individual key-value pair assigned to the session of more than 1KB although I'd want to be careful that there aren't performance implications from adding a step like this to session writes.

I wonder if we'd get a similar benefit from more general monitoring of session sizes. I'd be interested to see 50th, 90th, 99th percentile session cookie sizes over time.  As a somewhat cross-cutting concern, it's something we could "budget" and alert if the total size is getting out of hand, which would also help us catch individual features contributing to the problem.